### PR TITLE
Record and verify WebAuthn RPIDs

### PR DESCRIPTION
**Pull request by codingllama**
Record and verify the credential RPID in `WebauthnDevice`, so we can detect and warn against accidental RPID changes.

RPID changes are not allowed by WebAuthn, so there's little Teleport can do to mitigate them.

Users that have only "invalid" WebAuthn devices will get ["invalid credentials"][1] errors on login. While a bit opaque, this is likely to lead to an user reset, which is the correct fix if only a few users are affected.

[1]: https://github.com/gravitational/teleport/blob/7a90a0ff9943e4f536736372dbdc90d324f143a2/lib/web/apiserver.go#L2180

### CODE REVIEW

**Comment by codingllama** [comment]
Tested manually in a cluster with a Yubi4 and Yubi5 registered (empty CredentialRpId to begin with).

**Review by tobiaszheller — APPROVED** [review]
lgtm, just wonder, maybe rpid change in cluster should be stored in audit log as system event? It will also simplify debugging, wdyt @codingllama ?

**Comment by codingllama** [comment]
@tobiaszheller yes, that would be nice. I won't do that here, but might chase it later. It's simple enough for "online" changes, teleport.yaml are a bit harder but I think it could be done.

File: lib/auth/webauthn/login.go [hunk 1]
```diff
@@ -83,6 +83,31 @@ func (f *loginFlow) begin(ctx context.Context, user string, passwordless bool) (
...
+				"User device %q/%q has unexpected RPID=%q, excluding from allowed credentials. "+
+				"RPID changes are not supported by WebAuthn, this is likely to cause permanent authentication problems for your users. "+
+				"Consider reverting the change or reset your users so they may register their devices again.",
+				user,
+				devices[i].GetName(),
+				webDev.CredentialRpId)
+
+			// "Cut" device from slice.
+			l := len(devices)
+			devices[i] = devices[l-1]
```
Comment (zmb3) [inline lib/auth/webauthn/login.go:104]: Nit: use `slices.Delete` instead?

**Review by zmb3 — APPROVED** [review]

File: lib/auth/webauthn/login.go (hunk 1 above)
Comment (codingllama) [reply lib/auth/webauthn/login.go]: Done.

**Pull request merged**